rules/PHPUnit100/Rector/MethodCall/PropertyExistsWithoutAssertRector.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\PHPUnit\PHPUnit100\Rector\MethodCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Name;
+use Rector\PHPUnit\NodeAnalyzer\IdentifierManipulator;
+use Rector\PHPUnit\NodeAnalyzer\TestsNodeAnalyzer;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @url https://github.com/sebastianbergmann/phpunit/issues/4601
+ *
+ * @see \Rector\PHPUnit\Tests\CodeQuality\Rector\MethodCall\PropertyExistsWithoutAssertRector\PropertyExistsWithoutAssertRectorTest
+ */
+final class PropertyExistsWithoutAssertRector extends AbstractRector
+{
+    /**
+     * @var array<string, string>
+     */
+    private const RENAME_METHODS_WITH_OBJECT_MAP = [
+        'assertClassHasStaticAttribute' => 'assertTrue',
+        'classHasStaticAttribute' => 'assertTrue',
+        'assertClassNotHasStaticAttribute' => 'assertFalse',
+    ];
+
+    public function __construct(
+        private readonly IdentifierManipulator $identifierManipulator,
+        private readonly TestsNodeAnalyzer $testsNodeAnalyzer
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Replace delited PHPUnit methods: assertClassHasStaticAttribute, classHasStaticAttribute and assertClassNotHasStaticAttribute by property_exists()',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+$this->assertClassHasStaticAttribute("Class", "property");
+$this->classHasStaticAttribute("Class", "property");
+$this->assertClassNotHasStaticAttribute("Class", "property");
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+$this->assertTrue(property_exists("Class", "property"));
+$this->assertTrue(property_exists("Class", "property"));
+$this->assertFalse(property_exists("Class", "property"));
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class];
+    }
+
+    /**
+     * @param MethodCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $map = self::RENAME_METHODS_WITH_OBJECT_MAP;
+        if (! $this->testsNodeAnalyzer->isPHPUnitMethodCallNames($node, array_keys($map))) {
+            return null;
+        }
+
+        if ($node->isFirstClassCallable()) {
+            return null;
+        }
+
+        $firstNode = new Arg($node->getArgs()[0]->value);
```

Comment:
args[0] and args[1] should be checked first with `isset()`, return null early if not exists.

Reply:
This is an impossible situation, but OK, it was fixed